### PR TITLE
feat(ha-node): added helm chart for ha node agent

### PR DESCRIPTION
**Pull request by mtzaurus**
HA Node agent is now deployed as a DaemonSet.

### CODE REVIEW

**Comment by mtzaurus** [comment]
bors try

File: chart/templates/mayastor/ha/ha-node-daemonset.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: {{ .Release.Name }}-agent-ha-node
+  labels:
+    openebs.io/engine: io-engine
```
Comment (tiagolobocastro) [inline chart/templates/mayastor/ha/ha-node-daemonset.yaml:6]: should this be "app: agent-ha-node" ?

File: chart/templates/mayastor/ha/ha-node-daemonset.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    rollingUpdate:
+      maxUnavailable: 1
+  minReadySeconds: 10
+  template:
+    metadata:
+      labels:
+        app: agent-ha-node
+        openebs.io/logging: "true"
+    spec:
+      hostNetwork: true
```
Comment (tiagolobocastro) [inline chart/templates/mayastor/ha/ha-node-daemonset.yaml:22]: Does it need to be on the host network?

File: chart/templates/mayastor/ha/ha-node-daemonset.yaml (hunk 1 above)
Comment (mtzaurus) [reply chart/templates/mayastor/ha/ha-node-daemonset.yaml:6]: Good point.

File: chart/templates/mayastor/ha/ha-node-daemonset.yaml (hunk 2 above)
Comment (mtzaurus) [reply chart/templates/mayastor/ha/ha-node-daemonset.yaml:22]: OK

**Review by tiagolobocastro — APPROVED** [review]

**Comment by mtzaurus** [comment]
bors merge

**Comment by mtzaurus** [comment]
borse merge

**Review by chriswldenyer — APPROVED** [review]
Looks reasonable

**Comment by mtzaurus** [comment]
bors merge

**Pull request merged**